commit a4bea10eca68152e84ffc4eaeb9d20ec2ac34664
Author: Patrick Jefferson <henj@hp.com>
Date:   Tue Oct 3 01:14:34 2006 -0700

    [PATCH] Allow ide_generic_all to be used modular and built in
    
    Allow ide/pci/generic to claim chipsets as a a module or when built-in.  It
    requires using "all_generic_ide" as a boot option.
    
    Signed-off-by: Patrick Jefferson <henj@hp.com>
    Acked-by: Alan Cox <alan@redhat.com>
    Closes-Bug: 7017
    Signed-off-by: Andrew Morton <akpm@osdl.org>
    Signed-off-by: Linus Torvalds <torvalds@osdl.org>

diff --git a/drivers/ide/pci/generic.c b/drivers/ide/pci/generic.c
index 78810ba..0cb7b9b 100644
--- a/drivers/ide/pci/generic.c
+++ b/drivers/ide/pci/generic.c
@@ -41,15 +41,8 @@
 
 static int ide_generic_all;		/* Set to claim all devices */
 
-#ifndef MODULE
-static int __init ide_generic_all_on(char *unused)
-{
-	ide_generic_all = 1;
-	printk(KERN_INFO "IDE generic will claim all unknown PCI IDE storage controllers.\n");
-	return 1;
-}
-__setup("all-generic-ide", ide_generic_all_on);
-#endif
+module_param_named(all_generic_ide, ide_generic_all, bool, 0444);
+MODULE_PARM_DESC(all_generic_ide, "IDE generic will claim all unknown PCI IDE storage controllers.");
 
 static void __devinit init_hwif_generic (ide_hwif_t *hwif)
 {
